convert2rhel/pkgmanager/handlers/dnf/callback.py

```diff
@@ -0,0 +1,292 @@
+            self.done_drpm += 1
+        else:
+            self.done_files += 1
+            self.done_size += size
+
+        message = None
+
+        if status:
+            # the error message, no trimming
+            if status is pkgmanager.callback.STATUS_DRPM and self.total_drpm > 1:
+                message = "(%d/%d) [%s %d/%d]: %s" % (
+                    self.done_files,
+                    self.total_files,
+                    self._STATUS_MAPPING[status],
+                    self.done_drpm,
+                    self.total_drpm,
+                    package,
+                )
+                message = "%s - %s" % (message, err_msg)
+            else:
+                message = "(%d/%d) [%s]: %s" % (
+                    self.done_files,
+                    self.total_files,
+                    self._STATUS_MAPPING.get(status, "Unknown"),
+                    package,
+                )
+        else:
+            if self.total_files > 1:
+                message = "(%d/%d): %s" % (self.done_files, self.total_files, package)
+
+        if message:
+            loggerinst.info(message)
+
+
+class TransactionDisplayCallback(pkgmanager.TransactionDisplay):
+    """Transaction display callback for DNF transaction."""
+
+    def __init__(self):
+        """Constructor for the transaction display progress in DNF."""
+        super(TransactionDisplayCallback, self).__init__()
+        self.last_package_seen = None
+
+    def progress(self, package, action, ti_done, ti_total, ts_done, ts_total):
+        """Process and output the RPM operations in the transaction.
+
+        :param package: Package being processed in the transaction.
+        :type package: `dnf.package.Package`
+        :param action: The type of the action being used for the current
+            `package`. Only values that came from :func:`rpmtrans.TransactionDisplay.action.keys()` are valid.
+        :type action: int
+        :param ti_done: How much work was already been done in the transaction.
+        :type ti_done: int
+        :param ti_total: How much work in total is present in the transaction.
+        :type ti_total: int
+        :param ts_total: How much transactions are present in the transaction set.
+        :type ts_total: int
+        :param ts_done: Number, in order, of the current transaction in the transaction set.
+        :type ts_done: int
+        """
+        # We don't have any package or action (actually, it's probably that it
+        # will be all empty), let's just return earlier.
+        if action is None or package is None:
+            loggerinst.debug("No action or package was provided in the callback.")
+            return
+
+        # We convert the package here to a str because we just stand a
+        # normal-standard str comparision rather than what is implemented in
+        # `yum.sqlitesack.YumAvailablePackageSqlite`, as this class does
+        # version comparision and a bunch of other stuff. We don't care about
```

Comment:
```suggestion
        # Here we convert the package to a str because we just need to do a
        # standard str comparision rather than what is implemented in
        # `yum.sqlitesack.YumAvailablePackageSqlite` (that class does
        # version comparison and a bunch of other stuff). We don't care about
```